File: pset1/calculator.c
#include <stdio.h>
#include <stdlib.h>

//function declarations
int takesNoArgs(char op);
double processOp(double amt, char op, double val);
double processPlus(double amt, double val);
double processMinus(double amt, double val);
double processTimes(double amt, double val);
double processDivide(double amt, double val);
double processPrint(double amt);
static float processClear();

int main(int argc, char **argv){

  double amt = 0.0;
  char op;
  double val;
  char temp;

  printf("CS 2271 Calculator\n");
  
  while (1){
    op = getchar();
    if (op == EOF) break;
    if (op == 'Q') break;
 
    if (takesNoArgs(op)) {
	val = 0.0;
    }
    else {scanf("%lf", &val);}
    
    temp = getchar();
    while (temp != '\n'){
	temp = getchar();
    }
    amt = processOp(amt, op, val);
  }
  printf("Thank you and goodbye!\n");
}

int takesNoArgs(char op) { 
    return ((op == 'C') || (op == '='));
}

double processOp(double amt, char op, double val) {
    if (op == '+')
        return processPlus(amt, val);
    else if (op == '-')
        return processMinus(amt, val);
    else if (op == '*')
        return processTimes(amt, val);
    else if (op == '/')
        return processDivide(amt, val);
    else if (op == '=')
        return processPrint(amt);
    else if (op == 'C')
        return processClear();
    else {
        printf("Bad Operator \n");
        return amt;
    }
}

double processPlus(double amt, double val) {
    return amt + val;
}

double processMinus(double amt, double val) {
    return amt - val;
}

double processTimes(double amt, double val) {
    return amt * val;
}

double processDivide(double amt, double val) {
    return amt / val;
}

double processPrint(double amt) {
    printf("%f \n", amt);
    return amt;
}

static float processClear() {
    return 0;
}

File: pset1/Calculator.java
import java.io.*;

class Calculator {
    private static Reader r = new InputStreamReader(System.in);
    private static BufferedReader br = new BufferedReader(r);
    
    public static void main(String[] args) {
        float amt = 0.0F;  // the current value of the calculator
        char op;           // the operation to be performed
        float val;         // the value of the operation
        
        System.out.println("CS 2271 Calculator");
        while (true) {
            // Read an input line. The first char of each line is the operation.
            // Arithmetic operations are followed by whitespace and a number.
            try {
                int ic = br.read();
                if (ic < 0) break;   //EOF
                op = (char) ic;
                if (op == 'Q') break;
                
                String s = br.readLine();
                if (takesNoArgs(op))
                    val = 0.0F;
                else
                    val = Float.parseFloat(s);
            }
            catch (IOException e) {  // abort on any IO exception
                throw new RuntimeException("Something bad happened");
            }
            
            // Call a fn to process the op and its val.
            amt = processOp(amt, op, val);
        }
        System.out.println("Thank you and goodbye!");
    }
    
    static boolean takesNoArgs(char op) {
        return (op == 'C') || (op == '=');
    }
    
    static float processOp(float amt, char op, float val) {
        if (op == '+')
            return processPlus(amt, val);
        else if (op == '-')
            return processMinus(amt, val);
        else if (op == '*')
            return processTimes(amt, val);
        else if (op == '/')
            return processDivide(amt, val);
        else if (op == '=')
            return processPrint(amt);
        else if (op == 'C')
            return processClear();
        else {
            System.out.println("Bad Operator");
            return amt;
        }
    }
    
    static float processPlus(float amt, float val) {
        return amt + val;
    }
    
    static float processMinus(float amt, float val) {
        return amt - val;
    }
    
    static float processTimes(float amt, float val) {
        return amt * val;
    }
    
    static float processDivide(float amt, float val) {
        return amt / val;
    }
    
    static float processPrint(float amt) {
        System.out.println(amt);
        return amt;
    }
    
    static float processClear() {
        return 0;
    }
}
